### PR TITLE
Change how parse errors are reported.

### LINKED ISSUE #101: Analyzers report weird before protocol has been confirmed
Currently we often emit weird log entries if we see parse errors during explorative processing in `PacketAnalyzer::AnalyzePacket` and `ProtocolAnalyzer::[Process|Finish]`] (we also report weirds from `FileAnalyer` which might be different). If such reporting happens during DPD before confirmation parse errors are to be expected. Reporting them to the weird log on the other hand makes it look like there is a problem which needs to be investigated; we also make it hard to distinguish genuine parse errors (after DPD confirmation) from ones during DPD.

We should look how we can reduce weirds during DPD. We could e.g., track whether an analyzer has confirmed a protocol and suppress or redirect during DPD.

### PR DESCRIPTION
**Pull request by rsmmr**
Closes #101.

This is currently on top of #44.

This also adds new `zeek::weird()` function to report a Zeek-side weird.

### CODE REVIEW

**Comment by rsmmr** [comment]
Waiting to see if https://github.com/zeek/zeek/pull/2043 will make it into 4.2.x.

**Comment by bbannier** [comment]
@rsmmr anything I can do to get this unstuck?

**Comment by rsmmr** [comment]
ket me take a look, need to remind myself what's going on here

File: src/runtime-support.cc [hunk 1]
```diff
@@ -251,22 +251,40 @@ void rt::confirm_protocol() {
...
+
+    if ( const auto x = std::get_if<cookie::ProtocolAnalyzer>(cookie) )
+        x->analyzer->Weird(id.c_str(), addl.data());
+    else if ( const auto x = std::get_if<cookie::FileAnalyzer>(cookie) )
+        ::zeek::reporter->Weird(x->analyzer->GetFile(), id.c_str(), addl.data());
+    else if ( const auto x = std::get_if<cookie::PacketAnalyzer>(cookie) ) {
+        ::spicy::zeek::compat::PacketAnalyzer_Weird(x->analyzer, id.c_str(), x->packet, addl.data());
+    }
+    else
+        throw ValueUnavailable("neither $conn nor $file nor packet analyzer available for weird reporting");
```
Comment (bbannier) [inline src/runtime-support.cc:285]:
```suggestion
        throw ValueUnavailable("none of $conn, $file, or $packet available for weird reporting");
```

File: spicy/zeek.spicy [hunk 2]
```diff
@@ -12,6 +12,16 @@ public function confirm_protocol() : void &cxxname="spicy::zeek::rt::confirm_pro
 ## Triggers a DPD protocol violation for the current connection.
 public function reject_protocol(reason: string) : void &cxxname="spicy::zeek::rt::reject_protocol";
 
+## Reports a "weird" to Zeek. This should be used with similar semantics as in
+## Zeek: something quite unexpected happening at the protocol level, which however
+## does not prevent us from continuing to process the connection.
+##
+## id: the name of the weird, which (just like in Zeek) should be a *static*
+## string identifying the situation reported (``unexpected_command``).
```
Comment (bbannier) [inline spicy/zeek.spicy:20]:
```suggestion
## string identifying the situation reported (e.g., ``unexpected_command``).
```

File: src/protocol-analyzer.cc [hunk 3]
```diff
@@ -63,7 +64,9 @@ void ProtocolAnalyzer::Process(bool is_orig, int len, const u_char* data) {
         hilti::rt::context::CookieSetter _(&endp->cookie());
         endp->process(len, reinterpret_cast<const char*>(data));
     } catch ( const spicy::rt::ParseError& e ) {
-        reporter::weird(endp->cookie().analyzer->Conn(), e.what());
+        STATE_DEBUG_MSG(is_orig, hilti::rt::fmt("parse error, triggering analyzer violation: %s", e.what()));
```
Comment (bbannier) [inline src/protocol-analyzer.cc:67]: Let's also remove all the `reporter::weird` functions since they are now unused.

**Review by bbannier — APPROVED** [review]

**Pull request merged**